go/workflow.go

```diff
@@ -3,45 +3,41 @@ package app
 import (
 	"time"
 
+	"go.temporal.io/sdk/temporal"
 	"go.temporal.io/sdk/workflow"
-	"go.uber.org/multierr"
 )
 
-func BreakfastWorkflow(ctx workflow.Context) (err error) {
+func BreakfastWorkflow(ctx workflow.Context, parallel_compensations bool) (err error) {
 	options := workflow.ActivityOptions{
 		StartToCloseTimeout: time.Second * 5,
+		RetryPolicy:         &temporal.RetryPolicy{MaximumAttempts: 1},
 	}
 
 	ctx = workflow.WithActivityOptions(ctx, options)
 
+	var compensations Compensations
+
 	err = workflow.ExecuteActivity(ctx, GetBowl).Get(ctx, nil)
+	compensations.AddCompensation(PutBowlAway)
 	if err != nil {
 		return err
 	}
 
-	defer func() {
-		if err != nil {
-			errCompensation := workflow.ExecuteActivity(ctx, PutBowlAway).Get(ctx, nil)
-			err = multierr.Append(err, errCompensation)
-		}
-	}()
-
 	err = workflow.ExecuteActivity(ctx, AddCereal).Get(ctx, nil)
+	compensations.AddCompensation(PutCerealBackInBox)
 	if err != nil {
 		return err
 	}
 
+	err = workflow.ExecuteActivity(ctx, AddMilk).Get(ctx, nil)
+
 	defer func() {
```

Comment:
This defer should not be way down here, it should be up by where you declare the `compensations` variable. Otherwise, if the second activity failed, you would never run this.